relay-sampling/src/lib.rs

```diff
@@ -1287,7 +1330,9 @@ impl DynamicSamplingContext {
 
         let contexts = event.contexts.value()?;
         let context = contexts.get(TraceContext::default_key())?.value()?;
-        let Context::Trace(ref trace) = context.0 else { return None };
+        let Context::Trace(ref trace) = context.0 else {
```

Comment:
Sorry for causing conflicts through https://github.com/getsentry/relay/pull/2296. Once you merge or rebase master, this would be removed.